organisaatio-ui/src/components/Sivut/LomakeSivu/Koulutustoimija/OrganisaatioHistoriaLomake/OrganisaatioHistoriaLomake.tsx

```diff
@@ -1,118 +1,136 @@
 import * as React from 'react';
+import { useContext } from 'react';
 import styles from './OrganisaaatioHistoriaLomake.module.css';
 import YksinkertainenTaulukko from '../../../../Taulukot/YksinkertainenTaulukko';
 import useAxios from 'axios-hooks';
 import Spin from '@opetushallitus/virkailija-ui-components/Spin';
+import { LanguageContext } from '../../../../../contexts/contexts';
+import { Link } from 'react-router-dom';
 
 type organisaatioHistoriaProps = {
     oid?: string;
 };
 
 export default function OrganisaatioHistoriaLomake(props: organisaatioHistoriaProps) {
+    const { i18n, language } = useContext(LanguageContext);
     const [{ data: historia, loading: historiaLoading, error: historiaError }] = useAxios(
-        `/organisaatio/organisaatio/v4/1.2.246.562.10.69981965515/historia`
+        `/organisaatio/organisaatio/v4/${props.oid}/historia`
     );
-    console.log('hist', historia);
+    console.log('hist', historiaLoading, historia);
+    if (!historia || historiaLoading || historiaError) {
+        return <Spin />;
+    }
+
+    const renderLink = (a) => {
+        return (
+            <Link to={`/lomake/${a.oid}`}>
+                {(a.nimi[language] || a.nimi['fi'] || a.nimi['sv'] || a.nimi['en']) +
+                    (a.status !== 'AKTIIVINEN' ? '(' + i18n.translate('LABEL_' + a.status.toUpperCase()) + ')' : '')}
+            </Link>
+        );
+    };
+    const historiaSorter = (a, b) => {
+        return a.alkuPvm.localeCompare(b.alkuPvm);
+    };
+    const historiaMapper = (a, key) => {
+        return {
+            oid: a[key].oid,
+            nimiHref: renderLink(a[key]),
+            alkuPvm: a.alkuPvm,
+            loppuPvm: a.loppuPvm,
+            status: a[key].status,
+        };
+    };
```

Comment:
Just a note. These _could_ be outside component scope. Should ease up unit testing.